File: backend/src/main/java/com/vacanza/backend/dto/request/AutoCheckInRequestDTO.java
package com.vacanza.backend.dto;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AutoCheckInRequestDTO {

    @NotNull(message = "Latitude is required")
    private Double latitude;

    @NotNull(message = "Longitude is required")
    private Double longitude;

    @NotEmpty(message = "At least one candidate POI ID is required")
    private List<UUID> candidatePoiIds;
}

File: mobile/lib/features/checkin/data/models/auto_checkin_request_dto.dart
/// Request payload for `POST /checkins/auto`.
class AutoCheckinRequestDto {
  final double latitude;
  final double longitude;
  final List<String> candidatePoiIds;

  const AutoCheckinRequestDto({
    required this.latitude,
    required this.longitude,
    required this.candidatePoiIds,
  });

  Map<String, dynamic> toJson() => {
        'latitude': latitude,
        'longitude': longitude,
        'candidatePoiIds': candidatePoiIds,
      };
}
